app/frontend/wallet/constants.ts

```diff
@@ -115,3 +117,6 @@ export const RECOMMENDED_LEDGER_APP_VERSION = {
 }
 
 export const UNKNOWN_POOL_NAME = '<Unknown pool>'
+export const EPOCH_LENGTH = 432000
+export const EPOCHS_TO_REWARD_DISTRIBUTION = 4
+export const EPOCH_212_END = moment.utc('2020-08-23 21:44:00')
```

Comment:
These constants seem like related to the Shelley era, it would IMO make sense to move them to `NETWORKS` constant and import them from there. Or better, pass cryptoProvider.network to the function and get it from there.
I would rename the `EPOCH_212_END` at least to something like `firstRewardsDistributionTime` or something like that. It doesn't really say anything to the reader, that epoch 212 ended at this time.

Reply:
Refactored